src/ripple/app/paths/Flow.cpp

```diff
@@ -139,6 +147,8 @@ flow (
 
         if (f.ter == tesSUCCESS)
             strandSB.emplace (std::move (*f.sandbox));
+        else
+            removableOffers = std::move (f.removableOffers);
```

Comment:
Nice refactoring. I pushed a commit that took that idea (and modified it a little, please review).